tests/storages_tests/rdb_tests/test_with_server.py

```diff
@@ -0,0 +1,98 @@
+from multiprocessing import Pool
+import os
+from typing import Sequence
+from typing import Tuple
+
+import numpy as np
+import pytest
+
+import optuna
+
+_STUDY_NAME = "_test_multiprocess"
+
+
+def f(x: float, y: float) -> float:
+    return (x - 3) ** 2 + y
+
+
+def objective(trial: optuna.Trial) -> float:
+    x = trial.suggest_float("x", -10, 10)
+    y = trial.suggest_float("y", -10, 10)
+    trial.report(x, 0)
+    trial.report(y, 1)
+    trial.set_user_attr("x", x)
+    trial.set_system_attr("y", y)
+    return f(x, y)
+
+
+def run_optimize(args: Tuple[str, str]) -> None:
+    study_name = args[0]
+    storage = args[1]
+    # Create a study
+    study = optuna.create_study(study_name=study_name, storage=storage, load_if_exists=True,)
+    # Run optimization
+    study.optimize(objective, n_trials=20)
+
+
+@pytest.fixture
+def storage() -> str:
+    if "TEST_DB_URL" not in os.environ:
+        pytest.skip("This test requires TEST_DB_URL.")
+    storage_url = os.environ["TEST_DB_URL"]
+    try:
+        optuna.study.delete_study(_STUDY_NAME, storage_url)
+    except KeyError:
+        pass
+    return storage_url
+
+
+def _check_trials(trials: Sequence[optuna.trial.FrozenTrial]) -> None:
+    # Check trial states.
+    assert all((trial.state == optuna.trial.TrialState.COMPLETE for trial in trials))
+
+    # Check trial values and params.
+    assert all(("x" in trial.params for trial in trials))
+    assert all(("y" in trial.params for trial in trials))
+    assert all(
+        np.isclose(
+            [trial.value for trial in trials],
+            [f(trial.params["x"], trial.params["y"]) for trial in trials],
+            atol=1e-4,
+        )
+    )
+
+    # Check intermediate values.
+    assert all((len(trial.intermediate_values) == 2 for trial in trials))
+    assert all((trial.params["x"] == trial.intermediate_values[0] for trial in trials))
+    assert all((trial.params["y"] == trial.intermediate_values[1] for trial in trials))
+
+    # Check attrs.
+    assert all((trial.user_attrs["x"] == trial.params["x"] for trial in trials))
+    assert all((trial.system_attrs["y"] == trial.params["y"] for trial in trials))
+
+
+def test_many_trials(storage: str) -> None:
```

Comment:
More of a question but what kinds of problem is this test case trying to find with the large number of trials (that's not covered by existing unit tests)? I first thought that `tests-rdbstorage` was intended for distributed/parallel optimization but I understand that's not the case.

Reply:
I think this test becomes useful after [this suggestion](https://github.com/optuna/optuna/pull/1497/files#r454069756) is merged. However, I also agree with that we don't need such many trials to test storage's behavior.

Reply:
I thought study with many trials causes some issues, but I may get confused with performance measurement with a wall-time clock. I renamed the method to `test_loaded_study` and reduce the number of trials in https://github.com/optuna/optuna/pull/1497/commits/c8a3cbf4fe4ca4cad6065286fbce51e486581734.

Reply:
I'm wondering if this isn't covered by `test_multiprocess` already where a study is re-created using `load_study`.

Reply:
I think we can distinguish whether test failures are caused by multi-worker related issue or worker-database communications when they occur.
Additionally, we anyway need to run a study with a single worker somewhere before `tes_multiprocess` or add a CLI command to .circleci, otherwise multi-worker test fails because multiple workers try to create tables at the same time.

Reply:
Ah, that makes sense, thanks. 

As for the table creation, we might want to leave a comment to avoid future breakage (e.g. just placing the function definitions in a different order might break)?

Reply:
I agree that we'd better leave a comment.

Reply:
That sounds reasonable. I added a comment on table creation in https://github.com/optuna/optuna/pull/1497/commits/3d38a589305f394b13acd7813a3298960090e937 .
>     # Please create the tables by placing this function before the multi-process tests.